test/Microsoft.AspNetCore.Mvc.IntegrationTests/ModelBindingTestHelper.cs

```diff
@@ -151,7 +153,9 @@ public static IServiceProvider GetServices(Action<MvcOptions> updateOptions = nu
             var serviceCollection = new ServiceCollection();
             serviceCollection.AddAuthorization();
             serviceCollection.AddSingleton(new ApplicationPartManager());
-            serviceCollection.AddMvc();
+            serviceCollection
+                .AddMvc()
+                .SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
```

Comment:
This made surprisingly little difference. Updated a few tests to compensate and added one or two overrides.